geonode/layers/views.py

```diff
@@ -578,11 +541,9 @@ def dataset_metadata(
             la.featureinfo_type = form["featureinfo_type"]
             la.save()
 
-        if new_poc is not None or new_author is not None:
-            if new_poc is not None:
-                layer.poc = new_poc
-            if new_author is not None:
-                layer.metadata_author = new_author
+        # update contact roles
+        layer.set_contact_roles_from_metadata_edit(dataset_form)
```

Comment:
I see that the method `set_contact_roles_from_metadata_edit` return a `failed=True` if something went wrong.
If for any reason is not possible to set the contact roles, isn't it better to make the save fail?
even something like:

```python
is_failed = layer.set_contact_roles_from_metadata_edit(dataset_form)
if is_failed:
    raise Exception("Error during the contact role assignment")
```